example/main.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"flag"
+	"log"
+	"time"
+
+	sriovv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	sriovNamespaces "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/namespaces"
+	metallbv1beta1 "github.com/metallb/metallb-operator/api/v1beta1"
+	"github.com/openshift-kni/k8sreporter"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func main() {
+	kubeconfig := flag.String("kubeconfig", "", "the kubeconfig path")
+	testName := flag.String("testname", "test", "the file name used for the report")
+	reportPath := flag.String("path", "./report", "the root path used for the report")
+
+	flag.Parse()
+
+	// When using custom crds, we need to add them to the scheme
+	addToScheme := func(s *runtime.Scheme) {
+		sriovv1.AddToScheme(s)
+		metallbv1beta1.AddToScheme(s)
+	}
+
+	namespacesToDump := map[string]bool{
+		sriovNamespaces.Test: true,
+		"metallb-system":     true,
+	}
+
+	// The list of CRDs we want to dump
+	crds := []k8sreporter.CRData{
+		{Cr: &sriovv1.SriovNetworkNodePolicyList{}},
+		{Cr: &sriovv1.SriovNetworkList{}},
+		{Cr: &sriovv1.SriovNetworkNodePolicyList{}},
+		{Cr: &sriovv1.SriovOperatorConfigList{}},
+		{Cr: &metallbv1beta1.MetalLBList{}},
+	}
+
+	// The namespaces we want to dump resources for (including pods and pod logs)
+	dumpNamespace := func(ns string) bool {
```

Comment:
WDYT about having something similar exportable - receives a map and returns this func. That way the user doesn't have to rewrite it.

Reply:
fair enough, so I'd add it to the README as well.